test/test.c

```diff
@@ -4969,9 +4969,11 @@ _recvTestString(natsConnection *nc, natsSubscription *sub, natsMsg *msg,
         }
         case 4:
         {
-            arg->status = natsConnection_PublishString(nc,
+            arg->status = natsConnection_Publish(nc,
                                                        natsMsg_GetReply(msg),
-                                                       arg->string);
+                                                       natsMsg_GetData(msg),
+                                                       natsMsg_GetDataLength(msg));
+            arg->sum++;
```

Comment:
No, this is wrong. The arg->sum is incremented by each thread doing the request (in _sendRequest) if the result is as expected.